core/templates/dev/head/domain/state_card/StateCardObjectFactory.js

```diff
@@ -17,114 +17,197 @@
  * card domain objects used in the exploration player.
  */
 
-oppia.factory('StateCardObjectFactory', [function() {
-  var StateCard = function(
-      stateName, currentParams, contentHtml, interactionHtml, interaction,
-      leadsToConceptCard, destStateName, inputResponsePairs) {
-    this._stateName = stateName;
-    this._currentParams = currentParams;
-    this._contentHtml = contentHtml;
-    this._interactionHtml = interactionHtml;
-    this._leadsToConceptCard = leadsToConceptCard;
-    this._destStateName = destStateName;
-    this._inputResponsePairs = inputResponsePairs;
-    this._interaction = interaction;
-  };
-
-  StateCard.prototype.getDestStateName = function() {
-    return this._destStateName;
-  };
-
-  StateCard.prototype.getStateName = function() {
-    return this._stateName;
-  };
-
-  StateCard.prototype.getInteraction = function() {
-    return this._interaction;
-  };
-
-  StateCard.prototype.getCurrentParams = function() {
-    return this._currentParams;
-  };
-
-  StateCard.prototype.getContentHtml = function() {
-    return this._contentHtml;
-  };
-
-  StateCard.prototype.getInteractionHtml = function() {
-    return this._interactionHtml;
-  };
-
-  StateCard.prototype.getOppiaResponse = function(index) {
-    return this._inputResponsePairs[index].oppiaResponse;
-  };
-
-  StateCard.prototype.getLeadsToConceptCard = function() {
-    return this._leadsToConceptCard;
-  };
-
-  StateCard.prototype.getInputResponsePairs = function() {
-    return this._inputResponsePairs;
-  };
-
-  StateCard.prototype.getLastInputResponsePair = function() {
-    if (this._inputResponsePairs.length === 0) {
+oppia.factory('StateCardObjectFactory', [
```

Comment:
It would be good if you had unit tests for this domain object.

Reply:
Done.